src/passes/OptimizeInstructions.cpp

```diff
@@ -340,6 +340,50 @@ struct OptimizeInstructions
           return replaceCurrent(sub);
         }
       }
+      {
+        // signed(x <= y - 1)  =>  signed(x < y)
```

Comment:
Hmm, this doesn't work if `y` is INT_MIN and `x` is something like 0 because the right hand side will wrap around, right?

Reply:
Why?
`INT_MIN <= -1` => `true`
`INT_MIN < 0` => `true`

Reply:
But I see problem when `x` and `y` equals to `INT_MIN`. In this case it's not equivalent.
However LLVM optimize this: https://godbolt.org/z/cP46cEe8z

Reply:
`0 <= INT_MIN - 1` => `true`

`0 < INT_MIN` => `false`

Reply:
I do wonder if it would be worth adding a flag to tell the optimizer that it can assume there will be no overflow, in which case this would be valid. My guess is that it probably wouldn't be worth it because for C/C++, LLVM will have already made these optimizations. I assume other languages will have well-defined overflow, so would not want to tell Binaryen to assume there is no overflow.